### PR TITLE
Bug 1583870 - Migrate lint and ui-test tasks

### PR DESCRIPTION
**Pull request by JohanLorenzo**
---

Same as https://github.com/mozilla-mobile/fenix/pull/5430

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: automation/taskcluster/helper/get-secret.py [hunk 1]
```diff
@@ -20,7 +22,17 @@ def write_secret_to_file(path, data, key, base64decode=False, json_secret=False,
...
 
 def fetch_secret_from_taskcluster(name):
-    secrets = taskcluster.Secrets({'baseUrl': 'http://taskcluster/secrets/v1'})
+    try:
+        secrets = taskcluster.Secrets({
+            # BaseUrl is still needed for tasks that haven't migrated to taskgraph yet.
+            'baseUrl': 'http://taskcluster/secrets/v1',
+        })
+    except taskcluster.exceptions.TaskclusterFailure:
+        # taskcluster library >=5 errors out when `baseUrl` is used
```
Comment (mitchhentges) [inline automation/taskcluster/helper/get-secret.py]: Oh, while we're in this migration period, are we using different versions of the `taskcluster` library (both `4` and `5`)?

File: taskcluster/ci/test/kind.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - taskgraph.transforms.job:transforms
+    - taskgraph.transforms.task:transforms
+
+job-defaults:
```
Comment (mitchhentges) [inline taskcluster/ci/test/kind.yml]: Is it worth having `job-defaults` if there's only one job?

File: automation/taskcluster/helper/get-secret.py (hunk 1 above)
Comment (JohanLorenzo) [reply automation/taskcluster/helper/get-secret.py]: Yes. The new docker image installs the latest `taskcluster` lib, while the old manually crafted images are stuck on v4.

File: taskcluster/ci/test/kind.yml (hunk 2 above)
Comment (JohanLorenzo) [reply taskcluster/ci/test/kind.yml]: Not really. I kept it here because of Fenix/R-B. Removed!

**Comment by JohanLorenzo** [comment]
UI test green at https://tools.taskcluster.net/groups/YxqNCJf2SpmbKyB8IYSiZg/tasks/EWv-LMs2Rq-IDPcKj8NNmA/runs/0/logs/public%2Flogs%2Flive_backing.log

**Pull request merged**